lib/newFileMenu.ts

```diff
@@ -0,0 +1,112 @@
+/**
+ * @copyright Copyright (c) 2021 John Molakvoæ <skjnldsv@protonmail.com>
+ *
+ * @author John Molakvoæ <skjnldsv@protonmail.com>
+ *
+ * @license AGPL-3.0-or-later
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as
+ * published by the Free Software Foundation, either version 3 of the
+ * License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program. If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+import logger from "./utils/logger"
+
+export enum FileType {
+	FILE = 'file',
+	FOLDER = 'folder',
+}
+
+export interface Entry {
+	/** Unique ID */
+	id: string
+	/** Translatable string displayed in the menu */
+	displayName: string
+	// Default new file name
+	templateName: string
+	/**
+	 * Either iconSvgInline or iconClass must be defined
+	 * Svg as inline string. <svg><path fill="..." /></svg>
+	 */
+	iconSvgInline?: string
+	/** Existing icon css class */
+	iconClass?: string
+	fileType: FileType
```

Comment:
@juliushaertl @PVince81 any of you know why we actually use that property?
It has been like that for years
https://github.com/nextcloud/server/blob/ac2bc2384efe3c15ff987b87a7432bc60d545c67/apps/files/js/newfilemenu.js#L204
https://github.com/nextcloud/server/blob/ac2bc2384efe3c15ff987b87a7432bc60d545c67/apps/files/js/newfilemenu.js#L110

Reply:
Seems to be a leftover from how we used to manage creation
https://github.com/owncloud/core/commit/d3861685045a006b47c68185223be21a6233802e#diff-4e16996e525fd1d3ef643308e0c2edbd87c0776afb54b426b3a21cad6fca1868L181

Reply:
I guess we can drop this.
The only thing we need is icon, text, new file text string, ID and handler, no?